DEVEL.md

```diff
@@ -1,7 +1,253 @@
-# Development
+# Instructions for Device Plugin Development and Maintenance
 
+Table of Contents
 
-## How to develop simple device plugins
+* [Day-to-day Development How to's](#day-to-day-development)
+   * [Get the Source Code](#get-the-source-code)
+   * [Build and Run Plugin Binaries](#build-and-run-plugin-binaries)
+   * [Build Container Images](#build-container-images)
+   * [Build Against a Newer Version of Kubernetes](#build-against-a-newer-version-of-kubernetes)
+   * [Work with Intel Device Plugins operator modifications](#work-with-intel-device-plugins-operator-modifications)
+   * [Publish a New Version of the Intel Device Plugins Operator to operatorhub.io](#publish-a-new-version-of-the-intel-device-plugins-operator-to-operatorhub.io)
+   * [Run E2E Tests](#run-e2e-tests)
+   * [Run Controller Tests with a Local Control Plane](#run-controller-tests-with-a-local-control-plane)
+* [How to Develop Simple Device Plugins](#how-to-develop-simple-device-plugins)
+    * [Logging](#logging)
+    * [Error conventions](#error-conventions)
+* [Checklist for new device plugins](#checklist-for-new-device-plugins)
+
+## Day-to-day Development How to's
+### Get the Source Code
+
+With `git` installed on the system, just clone the repository:
+
+```bash
+$ export INTEL_DEVICE_PLUGINS_SRC=/path/to/intel-device-plugins-for-kubernetes
+$ git clone https://github.com/intel/intel-device-plugins-for-kubernetes ${INTEL_DEVICE_PLUGINS_SRC}
+```
+
+### Build and Run Plugin Binaries
+
+With `go` development environment installed on the system, build the plugin:
+
+```bash
+$ cd ${INTEL_DEVICE_PLUGINS_SRC}
+$ make gpu_plugin
+```
+
+**Note:** All the available plugin build targets is roughly the output of `ls cmd`.
+
+To test the plugin binary on the development system, run as administrator:
+
+```bash
+$ sudo -E ${INTEL_DEVICE_PLUGINS_SRC}/cmd/dsa_plugin/gpu_plugin
```

Comment:
typo: **dsa_plugin**

Reply:
There are many instances in these instructions where plugin name is specified => use a variable (e.g. $PLUGIN_NAME) for that too?

Reply:
Moved to `<plugin-build-target>` and `<image-build-target>`